csrc/apis/java/detector.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#include "detector.h"
+
+#include "common.h"
+
+namespace mmdeploy {
+
+class JavaDetector {
+ public:
+  const char *java_package_name;
+  JavaDetector(const char *model_path, const char *device_name, int device_id, const char *package_name) {
+    auto status = mmdeploy_detector_create_by_path(model_path, device_name, device_id, &handle_);
+    java_package_name = package_name;
+    if (status != MM_SUCCESS) {
+      throw std::runtime_error("failed to create detector");
+    }
+  }
+  static int draw_fps(int w, int h, cv::Mat& rgb)
+  {
+      // resolve moving average
+      float avg_fps = 0.f;
+      {
+          static double t0 = 0.f;
+          static float fps_history[10] = {0.f};
+
+          double t1 = ncnn::get_current_time();
+          if (t0 == 0.f)
+          {
+              t0 = t1;
+              return 0;
+          }
+
+          float fps = 1000.f / (t1 - t0);
+          t0 = t1;
+
+          for (int i = 9; i >= 1; i--)
+          {
+              fps_history[i] = fps_history[i - 1];
+          }
+          fps_history[0] = fps;
+
+          if (fps_history[9] == 0.f)
+          {
+              return 0;
+          }
+
+          for (int i = 0; i < 10; i++)
+          {
+              avg_fps += fps_history[i];
+          }
+          avg_fps /= 10.f;
+      }
+
+      char text[32];
+      sprintf(text, "%dx%d FPS=%.2f", w, h, avg_fps);
+
+      int baseLine = 0;
+      cv::Size label_size = cv::getTextSize(text, cv::FONT_HERSHEY_SIMPLEX, 0.5, 1, &baseLine);
+
+      int y = 0;
+      int x = rgb.cols - label_size.width;
+
+      cv::rectangle(rgb, cv::Rect(cv::Point(x, y), cv::Size(label_size.width, label_size.height + baseLine)),
+                      cv::Scalar(255, 255, 255), -1);
+
+      cv::putText(rgb, text, cv::Point(x, y + label_size.height),
+                  cv::FONT_HERSHEY_SIMPLEX, 0.5, cv::Scalar(0, 0, 0));
+
+      return 0;
+  }
+
+  extern "C" {
+    JNIEXPORT jint JNI_OnLoad(JavaVM* vm, void* reserved){
+      __android_log_print(ANDROID_LOG_DEBUG,"JNITag","enter jni_onload");
+      JNIEnv* env = NULL;
+      jint result = -1;
+      // ATTENTION JNI_VERSION VALUE!
+      if((*vm)->GetEnv(vm,(void**)&env,JNI_VERSION_1_6)!= JNI_OK){
+        return result;
+      }
+      // 定义函数映射关系（参数1：java native函数，参数2：函数描述符，参数3：C函数）
+      const JNINativeMethod method[]={
+              {"detectDraw","(II[I)Z",(void*)detectDraw},
```

Comment:
I don't think `Draw` is supposed to be an API

Reply:
Why not define an API to `mmdeploy_detector_apply` and `mmdeploy_detector_release_result`?

Reply:
Why not define an API to `mmdeploy_detector_apply` and `mmdeploy_detector_release_result`?